src/UtilityFunctions.cpp

```diff
@@ -41,6 +41,12 @@ int sumGeneratedNumbers(std::unique_ptr<NumberGenerator> gen, int num)
 
 int sumNumbersFromGenerators(std::vector<GeneratorType> const & types, int num)
 {
+    double sum = 0;
```

Comment:
This function returns `int`. Generators also work on `int` numbers. You have unnecessary conversions here. The best option is `int sum = 0;`